tests/ibr/test_service.py

```diff
@@ -185,55 +169,57 @@ def test_ibr_cancel_registration(self):
 
     def test_ibr_get_registration_fields(self):
         with unittest.mock.patch(
-                'aioxmpp.ibr.RegistrationService.connect_and_send'
+                'aioxmpp.protocol.send_and_wait_for',
+                new=CoroutineMock()
         ) as mock1:
-            mock1.return_value = self.auxiliar2()
-            res = run_coroutine(self.s.get_registration_fields(TEST_PEER))
-
-            _, (iq, *_), _ = mock1.mock_calls[0]
-
-            self.assertIsInstance(
-                iq,
-                aioxmpp.IQ,
+            iq = aioxmpp.IQ(
+                type_=aioxmpp.IQType.GET,
+                payload=ibr_xso.Query()
             )
-
-            self.assertEqual(
-                iq.to,
-                TEST_PEER.bare().replace(localpart=None),
+            iq.payload.username = ''
+            mock1.return_value = iq
+            stream = aioxmpp.protocol.XMLStream(
+                to=TEST_PEER.domain,
+                features_future=asyncio.Future()
             )
+            res = run_coroutine(aioxmpp.ibr.get_registration_fields(stream))
 
-            self.assertEqual(
-                iq.type_,
-                aioxmpp.IQType.GET,
-            )
+            _, (_, iq, *_), _ = mock1.mock_calls[0]
+
+            iq = iq[0]
 
             self.assertIsInstance(
-                iq.payload,
-                ibr_xso.Query,
+                iq,
+                aioxmpp.IQ,
             )
 
             self.assertIsInstance(
                 res,
-                list,
+                ibr_xso.Query,
             )
 
             self.assertEqual(
-                res,
-                ['username'],
+                res.username,
+                '',
             )
 
     def test_ibr_register(self):
         password = "aaa"
         aux_fields = {"nick": "romeo's lover"}
         with unittest.mock.patch(
-                'aioxmpp.ibr.RegistrationService.connect_and_send'
+                'aioxmpp.protocol.send_and_wait_for',
+                new=CoroutineMock()
         ) as mock1:
-            mock1.return_value = self.auxiliar3()
-            res = run_coroutine(
-                self.s.register(TEST_PEER, password, aux_fields)
+            stream = aioxmpp.protocol.XMLStream(
```

Comment:
You are not really using the stream here. Replace it with ``unittest.mock.sentinel.stream`` (which is just a dummy object).

This reduces the coupling between the test code and the XML stream implementation.

Reply:
I'm using some attribute of the xmlstream inside the method register, specifically the attribute _to.

Reply:
You’re right. I made a new suggestion in the new review.